File: backend/routers/health.py
"""Health check."""
from fastapi import APIRouter

from backend.celery_app import broker_is_reachable
from backend.database import check_schema
from backend.deps import engine

router = APIRouter()


@router.get("/health")
def health() -> dict:
    db = check_schema(engine)
    status = "ok" if db.get("connected") and db.get("schema_ready") else "degraded"
    if not db.get("connected"):
        status = "error"
    return {
        "status": status,
        "database": db,
        "queue": {"broker_reachable": broker_is_reachable()},
    }

File: frontend/src/api/health.ts
import { apiJson } from './client'

export type HealthResponse = {
  status: string
  database: {
    connected: boolean
    schema_ready: boolean
    missing_tables?: string[]
    missing_columns?: Record<string, string[]>
    message?: string
    error?: string
  }
  queue?: {
    broker_reachable: boolean
  }
}

export function fetchHealth() {
  return apiJson<HealthResponse>('/api/health')
}
